flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartition.java

```diff
@@ -252,16 +258,19 @@ BufferAndBacklog pollBuffer() {
 			}
 
 			if (buffer == null) {
+				dataAvailableNotified = false;
```

Comment:
I guess the path of `buffer == null` was caused by duplicated notification for `LocalInputChannel`. If we already fixed the duplicated notification in this PR, can we remove this path for simple?